src/librustc_resolve/resolve_imports.rs

```diff
@@ -627,14 +627,16 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
             self.finalize_resolutions_in(module);
         }
 
-        let mut has_errors = false;
         let mut seen_spans = FxHashSet::default();
         let mut errors = vec![];
         let mut prev_root_id: NodeId = NodeId::from_u32(0);
-        for i in 0 .. self.r.determined_imports.len() {
-            let import = self.r.determined_imports[i];
+        let mut determined_imports = vec![];
+        mem::swap(&mut self.r.determined_imports, &mut determined_imports);
+        let mut indeterminate_imports = vec![];
+        mem::swap(&mut self.r.indeterminate_imports, &mut indeterminate_imports);
+
+        for import in determined_imports.into_iter().chain(indeterminate_imports.into_iter()) {
             if let Some(err) = self.finalize_import(import) {
```

Comment:
I'm not sure that `finalize_import` is going to always return `Some(error)` for indeterminate imports.
It seems plausible though, so it would be great to have an assert checking for that here.

Reply:
Adding an `assert` for `Some(err)` when `import` comes from `indeterminate_imports` fails _almost_ always.